spdx/v2/v2_2/document.go

```diff
@@ -98,27 +99,46 @@ func (d *Document) UnmarshalJSON(b []byte) error {
 
 	*d = Document(d2)
 
+	relationshipExists := map[string]bool{}
+	serializeRel := func(r *Relationship) string {
+		return fmt.Sprintf("%v-%v->%v", common.RenderDocElementID(r.RefA), r.Relationship, common.RenderDocElementID(r.RefB))
+	}
+
+	// index current list of relationships to ensure no duplication
+	for _, r := range d.Relationships {
+		relationshipExists[serializeRel(r)] = true
+	}
+
 	// build relationships for documentDescribes field
 	for _, id := range e.DocumentDescribes {
-		d.Relationships = append(d.Relationships, &Relationship{
+		r := &Relationship{
 			RefA: common.DocElementID{
 				ElementRefID: d.SPDXIdentifier,
 			},
 			RefB:         id,
 			Relationship: common.TypeRelationshipDescribe,
-		})
+		}
+
+		if !relationshipExists[serializeRel(r)] {
+			d.Relationships = append(d.Relationships, r)
+			relationshipExists[serializeRel(r)] = true
+		}
```

Comment:
With the above change (which I've read might use less memory?), this section would be:
```go
		r := Relationship{
			RefA: common.DocElementID{
				ElementRefID: d.SPDXIdentifier,
			},
			RefB:         id,
			Relationship: common.TypeRelationshipDescribe,
		}

		if _, ok := allRelationships[r]; !ok {
			d.Relationships = append(d.Relationships, &r)
			allRelationships[r] = nil
		}
```